### PR TITLE
Manual Resume Review Endpoint:  Build API (This PR addresses the Linear issue KON2-59 - Konectin V2)

### PR DESCRIPTION
**Pull request by naanmohammed**
# Add Manual Review Feature
**_This feature builds on the existing AWS Helper Function. No modifications were made to the helper function. Extensive testing is required. Please be very critical in your reviews._**

This pull request introduces a new feature for manual resume reviews. It includes a new MongoDB schema for manual reviews and an endpoint to submit a resume for manual review.

## Assumptions I made when creating the endpoint

1. The function assumes that the user submitting the review already exists in the database and can be found by their email address.

2. It assumes that the AWS S3 bucket name is set in the environment variables.

3. It assumes that the resume file is uploaded as part of the request (req.files.resume if using POSTMAN).

4. The function assumes that the interestedCompanies field, if provided, is a comma-separated string that can be split into an array.

5. It assumes that the server has sufficient permissions to upload files to the specified AWS S3 bucket.

## To test the new endpoint in Postman:

1. Create a new POST request to `http://localhost:3001/user/submit-manual-review` .

2. Set the request body to form-data.

3. Add the following key-value pairs:
   - firstName (text)
   - lastName (text)
   - email (text)
   - currentCity (text)
   - currentCountry (text)
   - countryCode (text)
   - phoneNumber (text)
   - desiredJobTitle (text)
   - desiredIndustry (text)
   - interestedCompanies (text) - comma-separated list
   - resume (file) - upload a PDF or Word document

4. Send the request.

Example successful response:
`
{
  "message": "Manual review request submitted successfully",
  "reviewId": "60f1234567890abcdef12345"
}
`
## Notes

- Please set AWS_BUCKET_NAME, AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY) when testing.
- The submitted resume file will be temporarily stored on the server, uploaded to S3, and then deleted from the server.

Please review and test thoroughly before merging.

### CODE REVIEW

File: controllers/user/resume.controllers.js [hunk 1]
```diff
@@ -273,3 +278,87 @@ exports.numberOfDownloadeResumes = async (req, res) => {
...
+  filename: function (req, file, cb) {
+    cb(null, Date.now() + path.extname(file.originalname))
+  }
+});
+const upload = multer({ storage: storage }).single('resume');
+
+exports.submitManualReview = async (req, res) => {
+  try {
+    if (!process.env.AWS_BUCKET_NAME) {
+      console.error("AWS_BUCKET_NAME is not set in environment variables");
```
Comment (njoanc) [inline controllers/user/resume.controllers.js:295]: remove this

File: controllers/user/resume.controllers.js [hunk 2]
```diff
@@ -273,3 +278,87 @@ exports.numberOfDownloadeResumes = async (req, res) => {
...
+  },
+  filename: function (req, file, cb) {
+    cb(null, Date.now() + path.extname(file.originalname))
+  }
+});
+const upload = multer({ storage: storage }).single('resume');
+
+exports.submitManualReview = async (req, res) => {
+  try {
+    if (!process.env.AWS_BUCKET_NAME) {
```
Comment (njoanc) [inline controllers/user/resume.controllers.js:294]: this condition is not needed here

File: controllers/user/resume.controllers.js [hunk 3]
```diff
@@ -273,3 +278,87 @@ exports.numberOfDownloadeResumes = async (req, res) => {
...
+      currentCity,
+      currentCountry,
+      countryCode,
+      phoneNumber,
+      desiredJobTitle,
+      desiredIndustry,
+      interestedCompanies
+    } = req.body;
+
+    if (!firstName || !lastName || !email || !currentCity || !currentCountry || !countryCode || !phoneNumber || !desiredJobTitle || !desiredIndustry || !req.files || !req.files.resume) {
```
Comment (njoanc) [inline controllers/user/resume.controllers.js:312]: No need for this since we have required:true in our model

**Review by njoanc — COMMENTED** [review]
please check my reviews

**Comment by konnectintech** [comment]
@naanmohammed Can you make this PR to v2-test

**Comment by jeanned-arc** [comment]
Hello Naan

Can you please reach out to Michael? He is the one who has to do PR.
Thank you

**Pull request closed without merging**